scripts/evaluation/quality_evaluation.py

```diff
@@ -0,0 +1,163 @@
+"""Script to evaluate dialogue quality using LLM.
+
+The script evaluates dialogue quality with regards to five aspects:
+- Recommendation relevance
+- Communication style
+- Fluency
+- Conversational flow
+- Overall satisfaction
+
+Each aspect is scored between 1 and 5, where the scores are described in a
+dedicated rubric. The scoring is done using a large language model.
+"""
+
+import argparse
+import json
+import os
+from collections import defaultdict
+from dataclasses import dataclass
+from statistics import mean, stdev
+from typing import Dict, List, Union
+
+from tqdm import tqdm
+
+from dialoguekit.core.dialogue import Dialogue
+from dialoguekit.participant.participant import DialogueParticipant
+from dialoguekit.utils.dialogue_reader import json_to_dialogues
+from scripts.evaluation.rubrics.quality_rubrics import QualityRubrics
+from usersimcrs.simulator.llm.interfaces.ollama_interface import (
+    OllamaLLMInterface,
+)
+
+
+@dataclass
+class QualityScore:
+    conversation_id: str
+    score: int
+    explanation: str = ""
+
+    def to_dict(self) -> Dict[str, Union[int, str]]:
+        """Converts the score to a dictionary."""
+        return {
+            "conversation_id": self.conversation_id,
+            "score": self.score,
+            "score_explanation": self.explanation,
+        }
+
+
+class QualityScoreEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, QualityScore):
+            return o.to_dict()
+        return super().default(o)
+
+
+def parse_args() -> argparse.Namespace:
+    """Parse command-line arguments.
+
+    Returns:
+        Parsed arguments.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--dialogues",
+        type=str,
+        required=True,
+        help="Path to the dialogues.",
+    )
+    parser.add_argument(
+        "--ollama_config",
+        type=str,
+        required=True,
+        help="Path to the OLLAMA config file.",
+    )
+    parser.add_argument(
+        "--output",
+        type=str,
+        help="(optional) Path to the output file.",
+    )
+    return parser.parse_args()
+
+
+def get_prompt(grading_rubric: QualityRubrics, dialogue: Dialogue) -> str:
+    """Prepares prompt given grading rubric and dialogue.
+
+    Args:
+        grading_rubric: Grading rubric for the aspect.
+        dialogue: Dialogue.
+
+    Returns:
+        Prompt comprising task definition, grading rubric, and dialogue.
+    """
+    prompt = "You are an evaluator and you need to judge how does the "
+    "ASSISTANT perform based on the following CONVERSATION HISTORY. Please "
+    "rate the ASSISTANT's performance based on the following GRADING RUBRIC.\n"
+    "\nCONVERSATION HISTORY:"
```

Comment:
It might be nice to put the prompts as file-level consts on top of the file, e.g., `_PROMPT_xxx` for the different parts.